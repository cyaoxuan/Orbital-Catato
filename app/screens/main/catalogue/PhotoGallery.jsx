import { useState } from "react";
import { FlatList, Image, View } from "react-native";
import { FAB, Portal, Provider, Text } from "react-native-paper";
import { getItemWidth }  from "../../../utils/calculateItemWidths";
import { useRoute } from "@react-navigation/native";

export default function PhotoGallery() {
    const [open, setOpen] = useState(false);

    const route = useRoute();
    const imageSize = getItemWidth(2, 8);

    return (
        <Provider>
            <View>
                <FlatList
                    ListHeaderComponent={() => <Text variant="headlineMedium">{route.params.name} Meowmories</Text>}
                    ListHeaderComponentStyle={{ alignItems: "center" }}
                    numColumns={2}
                    contentContainerStyle={{ justifyContent: "space-around" }}
                    data={route.params.photoURLs}
                    renderItem={({item, index}) => {
                        return (
                            <View key={index} testID="gallery-photo">
                                <Image style={{ height: imageSize, width: imageSize, resizeMode: "cover", margin: 8 }} 
                                source={item} />
                            </View>
                        );
                    }}
                />
                <Portal>
                    <FAB.Group testID="fab-group" 
                        style={{ position: "absolute", bottom: 10, right: 10 }}
                        open={open}
                        icon={open ? "close" : "plus"}
                        actions={[
                            {icon: "camera", label: "Camera", onPress: () => {}},
                            {icon: "image", label: "Gallery", onPress: () => {}}
                        ]}
                        onStateChange={() => setOpen((prev) => !prev)}
                    />
                </Portal>
            </View>
        </Provider>
    )
}