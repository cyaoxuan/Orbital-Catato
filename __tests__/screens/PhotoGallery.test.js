import { cleanup, fireEvent, render } from "@testing-library/react-native";
import PhotoGallery from "../../app/screens/main/catalogue/PhotoGallery";

afterEach(cleanup)

jest.mock("@react-navigation/native", () => ({
    useRoute: jest.fn(() => ({
      params: {
        name: "Kitty",
        photoURLs: [
            require("../../assets/cats/cat-1-1.jpg"), 
            require("../../assets/cats/cat-1-2.jpg"),
            require("../../assets/cats/cat-1-3.jpg"),
            require("../../assets/cats/cat-1-4.jpg"),
            require("../../assets/cats/cat-1-5.jpg"),
            require("../../assets/cats/cat-1-6.jpg"),
            require("../../assets/cats/cat-1-7.jpg"),
        ],
      },
    })),
}));

describe("<PhotoGallery />", () => {
    it("renders photo gallery successfully", () => {
        const photoGallery = render(<PhotoGallery />);
        expect(photoGallery).toBeDefined();
    });

    it("renders photo gallery correctly", () => {
        const tree = render(<PhotoGallery />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders correct number of photos in gallery", () => {
        const { getAllByTestId } = render(<PhotoGallery />);
        const photos = getAllByTestId("gallery-photo");
        expect(photos).toHaveLength(7);
    });

    it("renders FAB group successfully", () => {
        const { getByTestId } = render(<PhotoGallery />);
        const fabGroup = getByTestId("fab-group");
        expect(fabGroup).toBeDefined();
    });
});