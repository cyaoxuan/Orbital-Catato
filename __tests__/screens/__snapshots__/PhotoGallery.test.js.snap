// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoGallery /> renders photo gallery correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <RCTScrollView
        ListHeaderComponent={[Function]}
        ListHeaderComponentStyle={
          {
            "alignItems": "center",
          }
        }
        contentContainerStyle={
          {
            "justifyContent": "space-around",
          }
        }
        data={
          [
            1,
            1,
            1,
            1,
            1,
            1,
            1,
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              {
                "alignItems": "center",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 28,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 36,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Kitty
               Meowmories
            </Text>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                testID="gallery-photo"
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 359,
                      "margin": 8,
                      "resizeMode": "cover",
                      "width": 359,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "bottom": 0,
            "justifyContent": "flex-end",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          {
            "bottom": 10,
            "position": "absolute",
            "right": 10,
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="none"
        style={
          {
            "backgroundColor": "rgba(255, 251, 254, 0.95)",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        pointerEvents="box-none"
        style={
          {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          pointerEvents="none"
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "marginBottom": 16,
                },
                {
                  "marginHorizontal": 24,
                },
              ]
            }
          >
            <View>
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 5,
                    "marginHorizontal": 16,
                    "marginVertical": 8,
                    "opacity": 0,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "transform": [
                      {
                        "translateY": -8,
                      },
                    ],
                  }
                }
                testID="card-container-outer-layer"
              >
                <View
                  accessibilityLabel="Camera"
                  accessibilityRole="button"
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "borderRadius": 5,
                      "elevation": 0,
                      "flex": undefined,
                      "paddingHorizontal": 12,
                      "paddingVertical": 6,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="card-container"
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                    testID="card"
                  >
                    <Text
                      index={0}
                      siblings={[]}
                      style={
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": 0.15,
                            "lineHeight": 24,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": "500",
                              "letterSpacing": 0.15,
                              "lineHeight": 24,
                            },
                            undefined,
                          ],
                        ]
                      }
                      total={1}
                    >
                      Camera
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(238, 232, 244)",
                  "borderRadius": 12,
                  "opacity": 0,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 4,
                    "width": 0,
                  },
                  "shadowOpacity": 0.15,
                  "shadowRadius": 8,
                  "transform": [
                    {
                      "translateY": -8,
                    },
                  ],
                }
              }
              testID="fab-container-outer-layer"
            >
              <View
                accessibilityRole="button"
                collapsable={false}
                pointerEvents="none"
                style={
                  {
                    "backgroundColor": "rgb(238, 232, 244)",
                    "borderRadius": 12,
                    "flex": undefined,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 3,
                  }
                }
                testID="fab-container"
              >
                <View
                  accessibilityLabel="Camera"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 12,
                      },
                    ]
                  }
                  testID="fab"
                >
                  <View
                    pointerEvents="none"
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        {
                          "borderRadius": 12,
                          "height": 40,
                          "width": 40,
                        },
                      ]
                    }
                    testID="fab-content"
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "justifyContent": "center",
                          },
                          {
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "bottom": 0,
                            "left": 0,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "transform": [
                              {
                                "rotate": "0deg",
                              },
                            ],
                          }
                        }
                        testID="cross-fade-icon-current"
                      >
                        <Text
                          accessibilityElementsHidden={true}
                          allowFontScaling={false}
                          importantForAccessibility="no-hide-descendants"
                          pointerEvents="none"
                          selectable={false}
                          style={
                            [
                              {
                                "color": "rgba(33, 0, 93, 1)",
                                "fontSize": 24,
                              },
                              [
                                {
                                  "lineHeight": 24,
                                  "transform": [
                                    {
                                      "scaleX": 1,
                                    },
                                  ],
                                },
                                {
                                  "backgroundColor": "transparent",
                                },
                              ],
                              {
                                "fontFamily": "material-community",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                        >
                          󰄀
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            pointerEvents="none"
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "marginBottom": 16,
                },
                {
                  "marginHorizontal": 24,
                },
              ]
            }
          >
            <View>
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 5,
                    "marginHorizontal": 16,
                    "marginVertical": 8,
                    "opacity": 0,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "transform": [
                      {
                        "translateY": -8,
                      },
                    ],
                  }
                }
                testID="card-container-outer-layer"
              >
                <View
                  accessibilityLabel="Gallery"
                  accessibilityRole="button"
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "borderRadius": 5,
                      "elevation": 0,
                      "flex": undefined,
                      "paddingHorizontal": 12,
                      "paddingVertical": 6,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="card-container"
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                    testID="card"
                  >
                    <Text
                      index={0}
                      siblings={[]}
                      style={
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "letterSpacing": 0.15,
                            "lineHeight": 24,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": "500",
                              "letterSpacing": 0.15,
                              "lineHeight": 24,
                            },
                            undefined,
                          ],
                        ]
                      }
                      total={1}
                    >
                      Gallery
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(238, 232, 244)",
                  "borderRadius": 12,
                  "opacity": 0,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 4,
                    "width": 0,
                  },
                  "shadowOpacity": 0.15,
                  "shadowRadius": 8,
                  "transform": [
                    {
                      "translateY": -8,
                    },
                  ],
                }
              }
              testID="fab-container-outer-layer"
            >
              <View
                accessibilityRole="button"
                collapsable={false}
                pointerEvents="none"
                style={
                  {
                    "backgroundColor": "rgb(238, 232, 244)",
                    "borderRadius": 12,
                    "flex": undefined,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 3,
                  }
                }
                testID="fab-container"
              >
                <View
                  accessibilityLabel="Gallery"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 12,
                      },
                    ]
                  }
                  testID="fab"
                >
                  <View
                    pointerEvents="none"
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        {
                          "borderRadius": 12,
                          "height": 40,
                          "width": 40,
                        },
                      ]
                    }
                    testID="fab-content"
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "justifyContent": "center",
                          },
                          {
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "bottom": 0,
                            "left": 0,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "transform": [
                              {
                                "rotate": "0deg",
                              },
                            ],
                          }
                        }
                        testID="cross-fade-icon-current"
                      >
                        <Text
                          accessibilityElementsHidden={true}
                          allowFontScaling={false}
                          importantForAccessibility="no-hide-descendants"
                          pointerEvents="none"
                          selectable={false}
                          style={
                            [
                              {
                                "color": "rgba(33, 0, 93, 1)",
                                "fontSize": 24,
                              },
                              [
                                {
                                  "lineHeight": 24,
                                  "transform": [
                                    {
                                      "scaleX": 1,
                                    },
                                  ],
                                },
                                {
                                  "backgroundColor": "transparent",
                                },
                              ],
                              {
                                "fontFamily": "material-community",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                        >
                          󰋩
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "marginBottom": 16,
              "marginHorizontal": 16,
              "marginTop": 0,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 8,
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
          testID="fab-group-container-outer-layer"
        >
          <View
            accessibilityRole="button"
            collapsable={false}
            pointerEvents="auto"
            style={
              {
                "backgroundColor": "rgba(234, 221, 255, 1)",
                "borderRadius": 16,
                "flex": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 3,
              }
            }
            testID="fab-group-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": false,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 16,
                  },
                ]
              }
              testID="fab-group"
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 16,
                      "height": 56,
                      "width": 56,
                    },
                  ]
                }
                testID="fab-group-content"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      {
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": [
                          {
                            "rotate": "0deg",
                          },
                        ],
                      }
                    }
                    testID="cross-fade-icon-current"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(33, 0, 93, 1)",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "material-community",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰐕
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
